makefile

```diff
@@ -0,0 +1,46 @@
+CC=g++
+CFLAGS=-std=c++17
+
+INCLUDECADMIUM=-I lib/cadmium/include
+
+#CREATE BIN AND BUILD FOLDERS TO SAVE THE COMPILED FILES DURING RUNTIME
+bin_folder := $(shell mkdir -p bin)
+build_folder := $(shell mkdir -p build/src build/test/subnet build/test/sender build/test/receiver)
+
+#TARGET TO RUN ONLY BIT SIMULATOR
+simulator: main.o ../data_structures/message.o
```

Comment:
you can rename "../data_structures/message.o" as it created confusion in the reader's mind. a user might think he has to navigate one directory back and go to data structures folder. this path actually doesn't exist. if you modify it here, make sure you change it in all the places in makefile